portality/view/doaj.py

```diff
@@ -74,8 +74,9 @@ def search():
 
 @blueprint.route("/search", methods=['POST'])
 def search_post():
+    """ Redirect a query from the box on the index page to the search page. """
     if request.form.get('origin') != 'ui':
-        abort(501)  # not implemented
+        abort(501)                                          # not implemented - we must receive searches from our own UI
```

Comment:
Sorry to be a pain, I don't know why I didn't see this before. I'd probably use a 400 here? (bad request) The problem is the client data being sent over is not "right".